app/src/settings/settings.rs

```diff
@@ -398,6 +400,44 @@ impl Component for SettingsPage {
                 self.group_list_expanded = !self.group_list_expanded;
                 true
             }
+            Msg::RegenerateToken => {
+                let app_link = ctx.props().app_link.clone();
+                wasm_bindgen_futures::spawn_local(async move {
+                    if let Err(e) = api_post::<()>((), "regenerate-token").await {
+                        alert(format!("Impossible de régénérer le token : {e}"));
+                    } else {
+                        // Force refresh of user info to update token
+                        <UserInfo as CachedData>::refresh(app_link);
+                    }
+                });
+                true
+            }
+            Msg::CopyIcs => {
+                let document = window().doc();
+                let value = document
+                    .get_element_by_id("ics-url")
+                    .and_then(|e| e.dyn_into::<HtmlInputElement>().ok())
+                    .map(|i| i.value())
+                    .unwrap_or_default();
+
+                let nav = window().navigator();
```

Comment:
```suggestion
                // We could use web-sys clipboard API but we are afraid it might panic on unsupported browsers. Could be changed in 2026
                let nav = window().navigator();
```